kit/src/components/message/mod.rs

```diff
@@ -460,3 +470,147 @@ pub fn markdown(text: &str) -> String {
 
     html_output
 }
+
+#[derive(Display)]
+pub enum IdentityCmd {
+    #[display(fmt = "GetIdentity")]
+    GetIdentity(DID),
+    #[display(fmt = "SentFriendRequest")]
+    SentFriendRequest(String, Vec<Identity>),
+}
+
+#[derive(Props, PartialEq)]
+pub struct IdentityMessageProps {
+    id: DID,
+}
+
+#[allow(non_snake_case)]
+pub fn IdentityMessage(cx: Scope<IdentityMessageProps>) -> Element {
+    let state = use_shared_state::<State>(cx)?;
```

Comment:
you can't use `State` in `kit`. you have to make a new `component` in `ui`.

Reply:
i am seeing quite a few state usages in `kit` though? why not?

Reply:
per design by @InfamousVague.

Reply:
i just dont really see a way to separate it from kit (or not have state) since its so deeply nested in kit components

Reply:
@sdwoodbury you have some approach ideas regarding this? only way i see it is like passing either passing some function or the component down the line but due to the depth might seem ugly.
or move the whole message thing to `ui` which isn't nice either

Reply:
@Flemmli97  sorry i forgot about this. let's make an exception for this case and go ahead with the current implementation.